player/playloop.c

```diff
@@ -1105,11 +1105,11 @@ static void handle_playback_time(struct MPContext *mpctx)
     } else if (mpctx->audio_status >= STATUS_PLAYING &&
                mpctx->audio_status < STATUS_EOF)
     {
-        mpctx->playback_pts = playing_audio_pts(mpctx);
+        mpctx->playback_pts = written_audio_pts(mpctx);
```

Comment:
That would only ever make a difference for an audio file with no video that has subtitles. It's true that the values are different though.

Reply:
That's why this is a RFC for people to test. We can always used the adjusted pts value for subtitles instead if it turns out that it has better behavior.

Reply:
I didn't look into this issue. What's the original issue? I recall it is to do with negative playback pts?

In video (+audio) mode it should be video pts, so non-negative. With playing audio adjusted to video pts.
In audio only mode it should be audio pts (written) + audio hardware delay for real in speaker audio. And to sync subtitles to current audio playing. 

So I'm confused, where is negative value coming from? I guess we cannot play audio from the future.

Reply:
`playing_audio_pts` subtracts the driver delay. That's how you can get negative numbers.

Reply:
I think it is valid that playback_pts is negative in audio mode, after all it will accurately show the delay until the actual 0 pts is played. All uses of this api should be aware of that and act accordingly. But I also agree we can just use written pts for things like watch later, where hardware delay makes only trouble.

Bonus question, what happens when we get and set playback position in audio mode? Does it roundtrip to the same position or we gonna jump back by the driver delay valie?

Reply:
This gets a bit messy because `get_current_time`/`get_playback_time` are indirectly exposed to users as properties and those are subject to the same negative pts situation. I think we agree this is probably not useful for anyone. If someone is really interested in that, we could add another property later.

Here's another possibility that is maybe less dramatic. Keep saving the playing_audio_pts in playback_pts as before but instead modify `get_playback_time` (maybe rename it to something more accurate) so the only negative value it can return is MP_NOPTS_VALUE. The function already has logic for a seeking edgecase. Use that for watch later instead of `get_current_time` and the properties already use `get_playback_time` so it should prevent negative values there.

Reply:
> so the only negative value it can return is MP_NOPTS_VALUE.

I kinda like this idea, mark negative values as clear no playing, regardless of what hardware delay is. And internally of course other things should still use real value.